core/trino-main/src/test/java/io/trino/sql/planner/optimizations/TestAddLocalExchangesForTaskScaleWriters.java

```diff
@@ -0,0 +1,126 @@
+import static io.trino.sql.planner.assertions.PlanMatchPattern.tableScan;
+import static io.trino.sql.planner.plan.ExchangeNode.Scope.LOCAL;
+import static io.trino.sql.planner.plan.ExchangeNode.Type.GATHER;
+import static io.trino.sql.planner.plan.ExchangeNode.Type.REPARTITION;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestAddLocalExchangesForTaskScaleWriters
+        extends BasePlanTest
+{
+    @Override
+    protected LocalQueryRunner createLocalQueryRunner()
+    {
+        Session session = testSessionBuilder()
+                .setCatalog("tpch")
+                .setSchema("tiny")
+                .build();
+        LocalQueryRunner queryRunner = LocalQueryRunner.create(session);
+        queryRunner.createCatalog("tpch", new TpchConnectorFactory(1), ImmutableMap.of());
+        queryRunner.createCatalog("mock_dont_report_written_bytes", createConnectorFactorySupportingReportingBytesWritten(false, "mock_dont_report_written_bytes"), ImmutableMap.of());
+        queryRunner.createCatalog("mock_report_written_bytes", createConnectorFactorySupportingReportingBytesWritten(true, "mock_report_written_bytes"), ImmutableMap.of());
+        return queryRunner;
+    }
+
+    private MockConnectorFactory createConnectorFactorySupportingReportingBytesWritten(boolean supportsWrittenBytes, String name)
+    {
+        return MockConnectorFactory.builder()
+                .withSupportsReportingWrittenBytes(supportsWrittenBytes)
+                .withGetTableHandle(((session, schemaTableName) -> null))
+                .withName(name)
+                .build();
+    }
+
+    @Test
+    public void testLocalScaledWriterDistributionWithSupportsReportingWrittenBytes()
+    {
+        // when task scale writers is enabled
+        assertDistributedPlan(
+                "CREATE TABLE mock_report_written_bytes.mock.test AS SELECT * FROM tpch.tiny.nation",
+                testSessionBuilder()
+                        .setSystemProperty(TASK_SCALE_WRITERS_ENABLED, "true")
+                        .setSystemProperty(SCALE_WRITERS, "false")
+                        .build(),
+                anyTree(
+                        exchange(LOCAL, REPARTITION, SCALED_WRITER_DISTRIBUTION,
+                                anyTree(
+                                        tableScan("nation")))));
+
+        // when task scale writers is disabled
+        assertDistributedPlanDoesNotMatch(
+                "CREATE TABLE mock_report_written_bytes.mock.test AS SELECT * FROM tpch.tiny.nation",
+                testSessionBuilder()
+                        .setSystemProperty(TASK_SCALE_WRITERS_ENABLED, "false")
+                        .setSystemProperty(SCALE_WRITERS, "false")
+                        .build(),
+                anyTree(
+                        exchange(LOCAL, GATHER, SCALED_WRITER_DISTRIBUTION,
```

Comment:
why there is still `SCALED_WRITER_DISTRIBUTION` even when `TASK_SCALE_WRITERS_ENABLED` is off?

Reply:
Here, we are checking `assertDistributedPlanDoesNotMatch`

Reply:
I've changed it from GATHER to REPARTITION